docs/doxygen/collectives.md

```diff
@@ -1,20 +1,150 @@
++ The UCC\_COLL\_TYPE\_BCAST operation moves the data from the src buffer of
+root participant to the dest buffer of all participants.
 
-\b Synchronized and Non-Synchronized Collectives: In the synchronized collective model, on entry, the participants cannot read or write to other participants without ensuring all participants have entered the collective operation. On the exit of the collective operation, the participants may exit after all participants have completed the reading or writing to the buffers.
++ The UCC\_COLL\_TYPE\_BARRIER synchronizes all participants of the collective
+operation. In this routine, first, each participant waits for all other
+participants to enter the operation. Then, once it learns the entry of all other
+participants into the operation, it exits the operation completing it locally.
 
-In the non-synchronized collective model, on entry, the participants can read or write to other participants. If the input and output buffers are defined on the team and RMA operations are used for data transfer, it is the responsibility of the user to ensure the readiness of the buffer. On exit, the participants may exit once the read and write to the local buffers are completed. 
++ The UCC\_COLL\_TYPE\_FANIN gathers the data from all participants in the team
+to the root participant. Each participant sends the same amount of data to the
+root. The src buffer on root and dest buffer on non-root
+participants is not defined.
 
-\b Buffer Ownership: The ownership of input and output buffers are transferred from the user to the library after invoking the ucc\_collective\_init routine and on return from the routine, the ownership is transferred back to the user. However, after invoking and returning from ucc\_collective\_post or ucc\_collective\_init\_and\_post routines, the ownership stays with the library and it is returned to the user, when the collective is completed. 
++ The UCC\_COLL\_TYPE\_FANOUT scatters the data from the root to the all
+participants in the team. The root sends the same of data to all participants.
+The dest buffer on root and src buffer on non-root participants is not defined.
+
++ In the UCC\_COLL\_TYPE\_GATHER operation, each participant of the collective
+operation sends data to the root participant. All participants send the same
+amount of data (block\_size) to the root. The size of the block is
+“dt\_elem\_size * count”. The total amount of data received by the root is equal
+to block_size * num\_participants. Here, the “count” represents the number of
+data elements. The
+"dt_elem_size" represents the size of the data element in bytes. The
+"num_participants" define the number of participants in the team. The data on
+the root is placed in the receive buffer ordered by the “ep” ordering. For
+example, if the participants’ endpoints are ordered as “ep\_a” to “ep\_n”, the
```

Comment:
How do we define "ordering" of non-contiguous EPs ? Say, ep is just a pointer - each process might have different ordering then (local pointers may differ).